src/api/RolesApi.ts

```diff
@@ -1,258 +1,485 @@
+ */
+const getRolesForUser = async (userId: number): Promise<IUserRoles> => {
+  if (process.env.NODE_ENV === "development") {
+    await sleep();
+    const response = getIUserRoles(
+      testRoles.filter((entry) => entry.User.Id === userId)
+    );
+    if (response.length === 1) {
+      return Promise.resolve(response[0]);
+    } else {
+      // If we returned 0 or more than 1 users worth of data
+      //  then default the user to having no roles
+      return Promise.resolve({} as IUserRoles);
+    }
+  } else {
+    try {
+      const response = getIUserRoles(
+        await spWebContext.web.lists
+          .getByTitle("Roles")
+          .items.filter(`User/Id eq '${userId}'`)
+          .select("Id", "User/Id", "User/Title", "User/EMail", "Title")
+          .expand("User")()
+      );
+      if (response.length === 1) {
+        return Promise.resolve(response[0]);
+      } else {
+        // If we didn't error from the API, but returned 0 or more than 1 users worth of data
+        //  then default the user to having no roles
+        return Promise.resolve({} as IUserRoles);
+      }
+    } catch (e) {
+      console.error(
+        `Error occurred while trying to fetch Roles for User with ID ${userId}`
+      );
+      console.error(e);
+      if (e instanceof Error) {
+        throw new ApiError(
+          e,
+          `Error occurred while trying to fetch Roles for User with ID ${userId}: ${e.message}`
+        );
+      } else if (typeof e === "string") {
+        throw new ApiError(
+          new Error(
+            `Error occurred while trying to fetch Roles for User with ID ${userId}: ${e}`
+          )
+        );
+      } else {
+        throw new ApiError(
+          undefined,
+          `Unknown error occurred while trying to fetch Roles for User with ID ${userId}`
+        );
+      }
     }
+  }
+};
 
-}
+/**
+ * Get the Roles of the current user.
+ *
+ * @returns The Roles for a given User in the form of the react-query results.  The data element is of RoleType[]
+ */
+export const useMyRoles = () => {
+  let userId: number;
+  if (process.env.NODE_ENV === "development") {
+    // TODO: Investigate making this a DEV variable along with other current user info, so we can mimic logging in as differnt users
+    userId = 1;
+  } else {
+    // userId moved inside statement determining if dev environment or not as was exiting without returning when not existing in dev
+    userId = _spPageContextInfo?.userId;
+  }
 
-export class RolesApiDev implements IRolesApi {
+  return useQuery({
+    queryKey: ["roles", userId],
+    queryFn: () => getRolesForUser(userId),
+    // We don't need to requery SharePoint for these
+    // Unless it is changing in our app -- and then we can
+    // have them invalidated, so it will re-query
+    staleTime: Infinity,
+    cacheTime: Infinity,
+    // Return just the RoleType[]
+    select: (data) => data.Roles.map((role) => role.Role),
+  });
```

Comment:
Agree -- initially I had it's own internal function and I simplified both to use the same -- makes even more sense to just call the one then, and eliminate the other all-together.  In the process of doing this, since we are now returning current user info straight from SP provided variable rather than a Net call, I updated UserApi to remove the function from the class to getCurrentUser, and instead created a hook that just transforms the SP var into the format we want.  Now the UserProvider doesn't have to wait either -- and can call the hook to get the users roles as the only thing to wait for.
Also fixed where my select for users was returning the whol object instead of just the RoleType[]